pylint/checkers/spelling.py

```diff
@@ -283,73 +301,43 @@ def close(self):
-                    # The lower cased version of word passed spell checking
-                    continue
-
-                # If we reached this far, it means there was a spelling mistake.
-                # Let's retry with the original work because 'unicode' is a
-                # spelling mistake but 'Unicode' is not
-                if self.spelling_dict.check(word):
-                    continue
-            except enchant.errors.Error:
-                self.add_message(
-                    "invalid-characters-in-docstring", line=line_num, args=(word,)
-                )
+            # Spell check is case sensitive by default, since 'unicode' is
+            # a spelling mistake, but 'Unicode' is not.
+            if self.hobj.spell(word):
                 continue
 
-            # Store word to private dict or raise a message.
+            # Don't raise a message if unknown words should be stored in
+            # the private dict file instead.
             if self.config.spelling_store_unknown_words:
-                if lower_cased_word not in self.unknown_words:
-                    self.private_dict_file.write("%s\n" % lower_cased_word)
-                    self.unknown_words.add(lower_cased_word)
-            else:
-                # Present up to N suggestions.
-                suggestions = self.spelling_dict.suggest(word)
-                del suggestions[self.config.max_spelling_suggestions :]
-
-                m = re.search(r"(\W|^)(%s)(\W|$)" % word, line)
-                if m:
-                    # Start position of second group in regex.
-                    col = m.regs[2][0]
-                else:
-                    col = line.index(word)
-
-                if starts_with_comment:
-                    col += 1
-                indicator = (" " * col) + ("^" * len(word))
-
-                self.add_message(
-                    msgid,
-                    line=line_num,
-                    args=(
-                        word,
-                        original_line,
-                        indicator,
-                        "'{}'".format("' or '".join(suggestions)),
-                    ),
+                if word not in self.unknown_words:
+                    self.private_dict_file.write(word + "\n")
+                    self.unknown_words.add(word)
+                continue
+
+            # Show an indicator under the occurrence of the word in the original line
+            m = re.search(r"(\W|^)({})(\W|$)".format(re.escape(word)), line)
+            col = m.regs[2][0]
+            indicator = (" " * col) + ("^" * len(word))
+
+            # Present up to N suggestions.
+            suggestions = ""
+            if self.config.max_spelling_suggestions > 0:
+                suggested_words = self.hobj.suggest(word)
+                del suggested_words[self.config.max_spelling_suggestions :]
+                suggestions = "Did you mean: '{}'?".format(
+                    "' or '".join(suggested_words)
```

Comment:
No need to delete the suggested words at line 332, just use `"".join(suggested_words[:self.config.max_spelling_suggestions]`. I don't know why the former code was doing that as well.